arangod/Utils/CollectionNameResolver.cpp

```diff
@@ -315,12 +300,10 @@ std::string CollectionNameResolver::localNameLookup(TRI_voc_cid_t cid) const {
 
   // DBserver case of a shard:
   if (collection && collection->planId() != collection->id()) {
-    try {
-      collection = ClusterInfo::instance()->getCollection(
+    collection = ClusterInfo::instance()->getCollectionNT(
         collection->vocbase().name(), std::to_string(collection->planId())
       );
-    }
-    catch (...) {
+    if (collection != nullptr) {
```

Comment:
isn't this inverting the previous logic?
I think it should be `collection == nullptr`